kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java

```diff
@@ -159,23 +200,23 @@ public int getConcurrency() throws SQLException {
 
   @Override
   public Date getDate(int columnIndex) throws SQLException {
-    Object obj = getObject(columnIndex);
+    final Object obj = getObject(columnIndex);
     if (obj == null) {
       return null;
     }
     if (obj instanceof Date) {
       return (Date) obj;
     }
-    try {
-      if (obj instanceof String) {
+    if (obj instanceof String) {
+      try {
         return Date.valueOf((String) obj);
+      } catch (Exception e) {
+        throw new SQLException(
+            "Illegal conversion to Date from column " + columnIndex + " [" + obj + "]", e);
       }
-    } catch (Exception e) {
-      throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to date: " + e, e);
```

Comment:
it's horrible that the previously reported error message does not contain the illegal data!